src/token_utilities.py

```diff
@@ -13,27 +15,49 @@ def add_padding(token_part):
     return token_part + '=' * (4 - len(token_part) % 4)
 
 def get_cache(cache_key):
-    print(cache.get(cache_key))
+    return cache.get(cache_key)
 
 def process_token(token):
-    payload = decode_access_token(token)
-    unique_name = payload.get('unique_name')
-    if not unique_name:
-        raise ValueError("Unique name not found in token payload")
-
-    scope = payload.get('scp')
-    time = datetime.now().strftime("%H:%M:%S")
-    today = date.today().strftime("%d-%m-%Y")
-
-    cache_key = 'user_data_' + unique_name
-    cache_data = {
-        "access_token": token,
-        "scope": scope,
-        "time": time,
-        "date": today,
-        "email_address": unique_name
-    }
-
-    cache.set(cache_key, cache_data)
-    get_cache(cache_key)
-    return cache_data
+    global opaque_token_count
+
+    parts = token.split('.')
+    if len(parts) == 3:
+        try:
+            payload = decode_access_token(token)
+            unique_name = payload.get('unique_name')
+            if not unique_name:
+                raise ValueError("Unique name not found in token payload")
+
+            scope = payload.get('scp')
+            time = datetime.now().strftime("%H:%M:%S")
+            today = date.today().strftime("%d-%m-%Y")
+
+            cache_key = 'user_data_' + unique_name
+            cache_data = {
+                "access_token": token,
+                "scope": scope,
+                "time": time,
+                "date": today,
+                "email_address": unique_name
+            }
+
+            cache.set(cache_key, cache_data)
+            return cache_data
+
+        except Exception as e:
+            opaque_token_count += 1
+            opaque_key = f'opaque_{opaque_token_count}'
+            cache.set(opaque_key, token)
+            raise e  
+    else:
```

Comment:
This else is redundant.

Raise an exception in the split in decode_access_token and handle it in the catch exception value error.